src/metabase/formatter.clj

```diff
@@ -125,7 +125,7 @@
 
         currency           (when currency?
                              (keyword (or currency "USD")))
-        integral?          (isa? (or effective_type base_type) :type/Integer)
+        integral?          (and (isa? (or effective_type base_type) :type/Integer) (integer? scale))
```

Comment:
I think it makes sense to do something like this:

```clojure
(and (isa? (or effective_type base_type) :type/Integer) (integer? (or scale 1))) ;; if scale is `nil`, assume it's value is 1 for this check
````

This way, we still let `integral?` be true for the case where a value IS an integer, but the scale value doesn't exist. This was the reason a test expecting the value of `9E3` suddenly saw `9.00E3`. 

Incorporating a change like I've suggested above should bring back that behaviour.